src/util/chains.ts

```diff
@@ -97,7 +131,52 @@ export const ID_TO_PROVIDER = (id: ChainId): string => {
       return process.env.JSON_RPC_PROVIDER_ARBITRUM_ONE!;
     case ChainId.ARBITRUM_RINKEBY:
       return process.env.JSON_RPC_PROVIDER_ARBITRUM_RINKEBY!;
+    case ChainId.POLYGON:
+      return process.env.JSON_RPC_PROVIDER_POLYGON!;
+    case ChainId.POLYGON_MUMBAI:
+      return process.env.JSON_RPC_PROVIDER_POLYGON_MUMBAI!;
     default:
       throw new Error(`Chain id: ${id} not supported`);
   }
 };
+
+function isMatic(
+  chainId: number
+): chainId is ChainId.POLYGON | ChainId.POLYGON_MUMBAI {
+  return chainId === ChainId.POLYGON_MUMBAI || chainId === ChainId.POLYGON;
+}
+
+class MaticNativeCurrency extends NativeCurrency {
+  equals(other: Currency): boolean {
+    return other.isNative && other.chainId === this.chainId;
+  }
+
+  get wrapped(): Token {
+    if (!isMatic(this.chainId)) throw new Error('Not matic');
+    return WRAPPED_NATIVE_CURRENCY[this.chainId]!;
+  }
+
+  public constructor(chainId: number) {
+    if (!isMatic(chainId)) throw new Error('Not matic');
+    super(chainId, 18, 'MATIC', 'Polygon Matic');
+  }
+}
+
+export class ExtendedEther extends Ether {
```

Comment:
can `ExtendedEther` and `MaticNativeCurrency` maybe be consolidated?? It seems like Ether is almost entirely rewritten anyway. Because then the name `ExtendedEther` is being used elswhere even when it could be MATIC. I'm just confused about these two being separate classes both used for MATIC

Reply:
will leave for now and rework when we move to sdk-core